### PR TITLE
Feature: Improved display of cases for prepositions and some verbs

### LINKED ISSUE #120: Translate cases in dictionary entries to the language that is selected for the interface
Прєдположимо, jа избрал jесм "Русский"  jак jезык интерфејса.

Jестли ja кликнем на "Склонение" (Declesion / Склоњенје), jа буду видєти "падеж: Имен., Вин., Род., Пред., ...". Але jестли jа напишем "from" в словник, jа буду видєти `iz (+Gen.)/из (+Gen.) [iz]`. Ja не разумем что jест `Gen`, ибо в русском интерфејсе туто буде `Род.`.

Повинно се не `iz (+Gen.)/из (+Gen.) [iz]` изставјати, а `iz (+Род.)/из (+Род.) [iz]` изставјати.

### PR DESCRIPTION
**Pull request by ru-danko**
Resolves #120 

Prepositions and some verbs in the database have an indication of the case with which they are used.
Added:
1. Display the case names in the selected interface language.
2. A new settings item for displaying case questions instead of case names.

image

### CODE REVIEW

File: src/services/dictionary.ts [hunk 1]
```diff
@@ -512,15 +514,23 @@ class DictionaryClass {
         to: string,
         flavorisationType: string,
         alphabets?: IAlphabets,
+        caseQuestions?: boolean
     ): ITranslateResult[] {
         return results.map((item) => {
             const isv = this.getField(item, 'isv');
-            const add = this.getField(item, 'addition');
+            const addArray = this.getField(item, 'addition').match(/\(.+?\)/) || [];
+            let add = addArray.find((elem) => elem.indexOf('(+') === -1) || '';
```
Comment (noomorph) [inline src/services/dictionary.ts:522]: Use `.startsWith`, please.

File: src/utils/getCaseTips/getCaseTips.ts [hunk 2]
```diff
@@ -11,6 +11,17 @@ export function getCaseTips(caseName: string, setName: string): string {
                 case 'V': return 'hej! ty! vy!';
                 default: return '';
             }
+        case 'nounShort': 
+            switch(caseName.slice(0,1).toUpperCase()) {
```
Comment (noomorph) [inline src/utils/getCaseTips/getCaseTips.ts:15]: Maybe `caseName[0].toUpperCase()` ?

File: src/utils/convertCases/convertCases.test.ts [hunk 3]
```diff
@@ -1,19 +1,49 @@
 import { convertCases } from 'utils/convertCases';
 
 describe('convertCases', () => {
-    it('should convert "+2" to +Gen.', () => {
-        expect(convertCases('+2')).toBe('+Gen.');
+    it('should convert "+1" to +Nom', () => {
```
Comment (noomorph) [inline src/utils/convertCases/convertCases.test.ts:4]: I'd go with something like this:

```ts
test.each([
    ['+1', '+Nom'],
    ['+2', '+Gen'],
    ['+3', '+Dat'],
    ['+4', '+Acc'],
    ['+5', '+Ins'],
    ['+6', '+Loc'],
    ['+7', '+Voc']
])('should convert "%s" to "%s"', (input, expected) => {
    expect(convertCases(input)).toBe(expected);
});

test.each([
    ['+Nom.', '+Nom'],
    ['+Gen.', '+Gen'],
    ['+Dat.', '+Dat'],
    ['+Acc.', '+Acc'],
    ['+Ins.', '+Ins'],
    ['+Loc.', '+Loc'],
    ['+Voc.', '+Voc']
])('should remove dot from "%s"', (input, expected) => {
    expect(convertCases(input)).toBe(expected);
});

test('should convert unknown abbreviations to an empty string', () => {
    expect(convertCases('+ABCD')).toBe('');
});
```

File: src/utils/convertCases/convertCases.ts [hunk 4]
```diff
@@ -1,8 +1,13 @@
...
+        .replace('+1', '+Nom')
+        .replace('+2', '+Gen')
+        .replace('+3', '+Dat')
+        .replace('+4', '+Acc')
+        .replace('+5', '+Ins')
+        .replace('+6', '+Loc')
+        .replace('+7', '+Voc')
+        .replace('.', ''); 
+    
+    return ['+Nom','+Gen','+Dat','+Acc','+Ins','+Loc','+Voc'].find((el) => caseInfo.indexOf(el) !== -1) ? caseInfo : '';
```
Comment (noomorph) [inline src/utils/convertCases/convertCases.ts:12]: Doesn't look good enough.

Better create a set outside, e.g. `const VALID_VALUES = new Set(['+Nom', '+Gen', ...])` and check `VALID_VALUES.has(caseInfo) ? caseInfo : ''`

File: src/components/ResultsCard/ResultsCard.tsx [hunk 5]
```diff
@@ -88,14 +89,17 @@ function renderOriginal(item, alphabets, index) {
                         lang={lang}
                     />
                 );
-            })} {item.ipa && <span className="ipa">[{item.ipa}]</span>}
+            })} 
+            {item.caseInfo && <> <span className="caseInfo">(+{t(`case${item.caseInfo.slice(2,-1)}`)})</span></>}
```
Comment (noomorph) [inline src/components/ResultsCard/ResultsCard.tsx:93]: Maybe it's worth to add a separate small render function for caseInfo?

File: src/utils/getCaseTips/getCaseTips.ts (hunk 2 above)
Comment (ru-danko) [reply src/utils/getCaseTips/getCaseTips.ts]: Done.

File: src/utils/convertCases/convertCases.test.ts (hunk 3 above)
Comment (ru-danko) [reply src/utils/convertCases/convertCases.test.ts]: Done.

File: src/utils/convertCases/convertCases.ts (hunk 4 above)
Comment (ru-danko) [reply src/utils/convertCases/convertCases.ts]: Done in a similar way.

File: src/utils/convertCases/convertCases.test.ts (hunk 3 above)
Comment (ru-danko) [reply src/utils/convertCases/convertCases.test.ts]: Done

File: src/components/ResultsCard/ResultsCard.tsx (hunk 5 above)
Comment (ru-danko) [reply src/components/ResultsCard/ResultsCard.tsx]: I don't want to make a function here yet...

File: src/components/Checkbox/Checkbox.scss [hunk 6]
```diff
@@ -1,8 +1,7 @@
 .checkbox {
   display: inline-flex;
-  margin: 0 8px 8px 0;
+  margin: 0 8px 4px 0;
```
Comment (noomorph) [inline src/components/Checkbox/Checkbox.scss:3]: What that was for?

**Review by noomorph — APPROVED** [review]

File: src/utils/convertCases/convertCases.ts [hunk 7]
```diff
@@ -1,8 +1,14 @@
-export function convertCases(add: string): string {
-    return add
-        .replace('+2', '+Gen.')
-        .replace('+3', '+Dat.')
-        .replace('+4', '+Acc.')
-        .replace('+5', '+Ins.')
-        .replace('+6', '+Loc.');
+export function convertCases(caseInfoRaw: string): string {
+    const caseList = ['+Nom','+Gen','+Dat','+Acc','+Ins','+Loc','+Voc'];
```
Comment (noomorph) [inline src/utils/convertCases/convertCases.ts:2]: Please move outside. I am not 100% sure if this is optimized and the array is not being created each time we enter the function.

File: src/utils/convertCases/convertCases.ts [hunk 8]
```diff
@@ -1,8 +1,14 @@
-export function convertCases(add: string): string {
-    return add
-        .replace('+2', '+Gen.')
-        .replace('+3', '+Dat.')
-        .replace('+4', '+Acc.')
-        .replace('+5', '+Ins.')
-        .replace('+6', '+Loc.');
+export function convertCases(caseInfoRaw: string): string {
```
Comment (noomorph) [inline src/utils/convertCases/convertCases.ts:1]:
```suggestion
const caseList = ['+Nom','+Gen','+Dat','+Acc','+Ins','+Loc','+Voc'];

export function convertCases(caseInfoRaw: string): string {
```

Comment (ru-danko) [reply src/utils/convertCases/convertCases.ts:3]: Thank you! Done.

**Review by noomorph — APPROVED** [review]

**Review by noomorph — APPROVED** [review]

**Pull request merged**